pngcanvas.py

```diff
@@ -258,14 +268,16 @@ def load(self, f):
             raise ValueError('Unsupported PNG format (filter_type={}; must be 0)'.format(filter_type))
         if interlace != 0:
             raise ValueError('Unsupported PNG format (interlace={}; must be 0)'.format(interlace))
-        if color_type != 6:
-            raise ValueError('Unsupported PNG format (color_type={}; must be 6)'.format(color_type))
+        if color_type not in (2, 6):
+            raise ValueError('Unsupported PNG format (color_type={}; must one of [2, 6])'.format(color_type))
 
         self.width = width
         self.height = height
         self.canvas = bytearray(self.bgcolor * width * height)
-        row_size = width * 4
+        bpp = 4 if color_type == 6 else 3
+        row_size = width * bpp
         step_size = 1 + row_size
+        rgba_row_size = width * 4
```

Comment:
Actually, I think you don't need this variable since you define `row_size` through `bpp`.